src/Functions/array/arrayLogicalFunction.cpp

```diff
@@ -0,0 +1,548 @@
+#include <Columns/ColumnArray.h>
```

Comment:
Comment related to this file in general. Please add in strategic places high-level comments what the code/algorithms is doing. For a casual reader, it is almost impossible to follow without tedious debugging to understand what is happening in this file.

(as an example, you could look at src/Functions/array/arrayFold.cpp which I commented recently after a severe bug was found in the initial implementation)